internal/storage/types/location_config.go

```diff
@@ -53,14 +54,21 @@ func (c *LocationConfig) UnmarshalJSON(blob []byte) error {
 		return err
 	}
 
+	errMultipleConfig := errors.New("multiple config values have been assigned")
 	switch {
-	// TODO: return error if we have more than one config assigned (mutually exclusive)
 	case types.S3 != nil:
 		c.Value = types.S3
 	case types.SFTPConfig != nil:
+		if reflect.ValueOf(c.Value).IsValid() {
```

Comment:
I think that we can still use a simple nil check here, right? If you disagree please show me what's the value added because I can't see it. This could be a starting point for us to understand it: https://go.dev/play/p/y2XQtTZ1jnQ.